nautobot/docs/development/getting-started.md

```diff
@@ -538,6 +544,44 @@ In cases where you haven't made any changes to the database (which is most of th
+
+Integration testing is much more involved, and builds on top of the foundation laid by unit testing. As a guiding principle, integration tests should be comprehensive, because they are the last mile to asserting that Nautobot does what it is advertised to do. Without integration testing, we have to do it all manually, and that's no fun for anyone!
+
+Running integrations tests requires the use of Docker at this time. They can be directly invoked using `nautobot-server test` just as unit tests can, however, a headless Firefox browser provided by Selenium is required. Because Selenium installation and setup is complicated, we have included a configuration for this to work out of the box using Docker. 
+
+The Selenium contianer is running a standalone, headless Firefox "web driver" browser that can be remotely controlled by Nautobot for use in integration testing.
+
+Before running integration tests, the `selenium` container must be running. If you are using the Docker Compose workflow, it is automatically started for you. For the Virtual Environment workflow, you must start it manually.
+
+| Docker Compose Workflow   | Virtual Environment Workflow      |
+|---------------------------|-----------------------------------|
+| (automatic)               | `invoke start --service selenium` |
+
+Integration tests are run using the `invoke integration_test` command.
+
+| Docker Compose Workflow   | Virtual Environment Workflow                                                                           |
+|---------------------------|--------------------------------------------------------------------------------------------------------|
+| `invoke integration_test` | `nautobot-server test --config=nautobot/core/tests/nautobot_config.py nautobot.core.tests.integration` |
+
+!!! info
+    The same arguments supported by `invoke unittest` are supported by `invoke integration_test`. The key difference being the dependency upon the Selenium container, and a different default test label. Additionally, you may also use `invoke integration-test` in the Virtual Environment workflow given that the container is running, and that the `INVOKE_NAUTOBOT_LOCAL=True` environment variable has been set.
+
+Unlike unit tests, where the tests live adjacent to each inner application within the Nautobot code, integration tests must only ever be defined in `nautobot.core.tests.integration`. The reason for this is that integration tests are designed to test the core application and all of its functionality in one place. Integration tests must never be added anywhere but `nautobot.core.tests.integration`. We never want to risk running the unit tests and integration tests at the same time. The isolation from each other is critical to a clean and managable continuous development cycle.
```

Comment:
I'm a bit unhappy with this constraint, tbh - I'd rather that we be able to define app-focused integration tests in `nautobot.*.tests.integration` as otherwise my concern is that over time `nautobot.core.tests.integration` will grow to be unmanageable.

Could we maybe consider one of the following alternatives?
1.  Move unit tests to `nautobot/<app>/tests/unit/` and have integration tests in `nautobot/<app>/tests/integration/`? (This would make for a nicer folder structure, IMHO, but is a bit trickier from a test case labeling/discovery perspective)
2. Move tests to `nautobot/tests/unit/<app>/` and `nautobot/tests/integration/<app>/`? (This would be easier for test discovery as we could run `nautobot-server test nautobot.tests.unit` and/or `nautobot-server test nautobot.tests.integration`. It'd also remove the need for a `NAUTOBOT_INTEGRATION_TEST` variable, I think?)

Reply:
I am incredibly partial to option 2 for every reason you stated. It's cleaner IMO, and obviates the need for `NAUTOBOT_INTEGRATION_TEST`. I do feel that would be a mess for this body of work and should be done later.